### ISSUE TITLE
Our routing is a dirty hack

### ISSUE THREAD

**Issue by spratt** [SteamedPears/Code-Review#98]
I wrote the simplest possible routing while learning to use node when I first wrote the app and it hasn't been touched since.  We may want to replace it with a micro-framework like https://github.com/gjritter/nerve/

Thoughts?

**Comment by cdelahousse** [comment]
We get routing for free with Connect.

**Comment by spratt** [comment]
I assume you mean that connect has a routing module.

**Comment by spratt** [comment]
Ah, it looks like connect is the router.  Wow, connect's documentation is terrible.

**Comment by spratt** [comment]
What does connect.use(...) do?!

**Comment by psimonyi** [comment]
But there's source code! Source code is better than documentation because it's more precise.

**Comment by spratt** [comment]
Looks like connect.use([route='/',]fn) adds routes to the router.

**Comment by spratt** [comment]
Yeah, we could rewrite about half the backend in a few lines using connect.

**Comment by spratt** [comment]
Let me try to _express_ my seething hatred for connect.

Remember what I said about documentation?  I take it back.  Their documentation is beyond terrible.  It's harmful.

See the documentation for the urlencoded middleware: http://www.senchalabs.org/connect/urlencoded.html

It's basically just the source for that middleware, right?  Wrong!  It's actually some older version of the source for that middleware.

See the actual source in their github repo, which matches what I got by running npm install connect: https://github.com/senchalabs/connect/blob/master/lib/middleware/urlencoded.js

See that line that checks if the request has a body, and if it doesn't, it doesn't parse it?  Yeah!  I just spent 2.5 hours tracking that down.

ARRRRRRRRRRRRRRGGGGGGGGGGGGHHHHHHHHHHHH
